File: zig/src/sort_benchmark.zig
const std = @import("std");
const Helper = @import("helper.zig").Helper;

pub const SortBenchmark = struct {
    allocator: std.mem.Allocator,
    helper: *Helper,
    data: std.ArrayListUnmanaged(i32),
    size_val: i64,
    result_val: u64,

    pub fn init(allocator: std.mem.Allocator, helper: *Helper, bench_name: []const u8) !SortBenchmark {
        const size_val = helper.configVal(bench_name, "size") orelse 100000;
        var self = SortBenchmark{
            .allocator = allocator,
            .helper = helper,
            .data = .{},
            .size_val = size_val,
            .result_val = 0,
        };
        try self.data.ensureTotalCapacity(allocator, @as(usize, @intCast(size_val)));
        helper.reset();
        for (0..@as(usize, @intCast(size_val))) |_| {
            const val = helper.nextInt(1_000_000);
            self.data.appendAssumeCapacity(val);
        }
        return self;
    }

    pub fn deinit(self: *SortBenchmark) void {
        self.data.deinit(self.allocator);
    }
};

File: rust/src/benchmarks/sort_benchmark.rs
use super::super::helper;
use crate::config_i64;

pub struct SortBenchmark {
    pub size_val: i64,
    pub data: Vec<i32>,
    pub result_val: u32,
}

impl SortBenchmark {
    pub fn new_base() -> Self {
        Self {
            size_val: 0,
            data: Vec::new(),
            result_val: 0,
        }
    }

    pub fn prepare(&mut self, class_name: &str) {
        if self.size_val == 0 {
            self.size_val = config_i64(class_name, "size");
            self.data.reserve(self.size_val as usize);
            for _ in 0..self.size_val {
                self.data.push(helper::next_int(1_000_000));
            }
        }
    }
}
